### PR TITLE
Add processing of NVHPC flags in linux-gnulibc1 file

### PR DESCRIPTION
**Pull request by jhendersonHDF**
Pull in NVHPC flags from `nvidia-flags`, `nvidia-fflags` and `nvidia-cxxflags`

### CODE REVIEW

**Comment by jhendersonHDF** [comment]
I'm expecting this will fail in the Autotools NVHPC action since, without #3800, it will add in `-fast` to the compile options from `nvidia-flags`, which implies `-O2`.

With these changes, I can see that Autotools builds with NVHPC now get `-c99 -Minform=warn -fast  -s   -O1` in `H5_CFLAGS`

**Comment by jhendersonHDF** [comment]
Looks like this is indeed now failing in the same places as CMake with the flags being pulled in correctly. I've disabled the testing in this PR and then will add re-enabling it back to #3800, though it appears that needs more work to pass testing.

File: .github/workflows/nvhpc-auto.yml [hunk 1]
```diff
@@ -67,11 +67,11 @@ jobs:
...
-      - name: Autotools Run Tests
-        run: |
-          export PATH=/opt/nvidia/hpc_sdk/Linux_x86_64/23.7/comm_libs/openmpi4/bin:/opt/nvidia/hpc_sdk/Linux_x86_64/23.7/compilers/bin:$PATH
-          make check -j
-        working-directory: ${{ runner.workspace }}/build
+#      - name: Autotools Run Tests
+#        run: |
+#          export PATH=/opt/nvidia/hpc_sdk/Linux_x86_64/23.7/comm_libs/openmpi4/bin:/opt/nvidia/hpc_sdk/Linux_x86_64/23.7/compilers/bin:$PATH
+#          make check -j
+#        working-directory: ${{ runner.workspace }}/build
```
Comment (derobins) [inline .github/workflows/nvhpc-auto.yml:74]: Why are the checks commented out?

**Review by byrnHDF — APPROVED** [review]

File: .github/workflows/nvhpc-auto.yml (hunk 1 above)
Comment (lrknox) [reply .github/workflows/nvhpc-auto.yml:74]: See previous comment.

**Review by lrknox — APPROVED** [review]

**Pull request merged**